src/main/java/org/spongepowered/mod/effect/particle/SpongeParticleHelper.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.effect.particle;
+
+import java.awt.Color;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.network.Packet;
+import net.minecraft.network.play.server.S2APacketParticles;
+import net.minecraft.util.EnumParticleTypes;
+
+import org.spongepowered.api.item.ItemBlock;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.inventory.ItemStack;
+
+import com.flowpowered.math.vector.Vector3d;
+import com.flowpowered.math.vector.Vector3f;
+import com.google.common.collect.Lists;
+
+public final class SpongeParticleHelper {
+
+    /**
+     * Gets the list of packets that are needed to spawn the particle effect at the position.
+     * This method should tries to minimize the amount of packets for a better performance and
+     * lower bandwidth use.
+     * 
+     * @param effect The particle effect
+     * @param position The position
+     * @return The packets
+     */
+    public static List<Packet> toPackets(SpongeParticleEffect effect, Vector3d position) {
```

Comment:
Is it necessary to create the packets in Sponge?
Most of this can be done using `net.minecraft.world.World#spawnParticle` (`WorldServer#func_180505_a` for the server impl)
Perhaps a tweak to the world class would be better than this complex packet creation